### PR TITLE
Issue #11719: Activate all group for pitest-blocks

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
#11719
Activating `ALL` group for `pitest-blocks`

Every mutator has been activated except `INLINE_CONSTS`

- Report with `ALL` group except `INLINE_CONSTS`: https://vyom-yadav.github.io/pitest-all-latest/pitest-blocks/index.html

### CODE REVIEW

File: .ci/pitest-suppressions/pitest-blocks-suppressions.xml [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    <lineContent>switch (ast.getType()) {</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>RightCurlyCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck$Details</mutatedClass>
+    <mutatedMethod>getDetails</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.RemoveSwitchMutator_3</mutator>
+    <description>RemoveSwitch 3 mutation</description>
+    <lineContent>switch (ast.getType()) {</lineContent>
```
Comment (nick-mancuso) [inline .ci/pitest-suppressions/pitest-blocks-suppressions.xml:144]: From https://pitest.org/quickstart/mutators/#EXPERIMENTAL_SWITCH:

>The switch mutator finds the first label within a switch statement that differs from the default label. It mutates the switch statement by replacing the default label (wherever it is used) with this label. All the other labels are replaced by the default one.

@Vyom-Yadav can you send a PR with hardcoded mutation for this one to make it easy for reviewers to understand this mutation and see the value in it.

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
Item:

File: .ci/pitest-suppressions/pitest-blocks-suppressions.xml (hunk 1 above)
Comment (Vyom-Yadav) [reply .ci/pitest-suppressions/pitest-blocks-suppressions.xml:144]: #11982 The definition of what the mutation does is not clear at all, went through the pitest source code to see what's happening. Number 3 here specifies the switch case to shift to the default branch (0-based indexing). From source code Javadoc:

>Remove switch statements. We get an array of labels to jump to, plus a default label. We change each label to the default label, thus removing it

Hardcoded mutation in #11982 is for value 4.
```XML
  <mutation unstable="false">
    <sourceFile>RightCurlyCheck.java</sourceFile>
    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck$Details</mutatedClass>
    <mutatedMethod>getDetails</mutatedMethod>
    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.RemoveSwitchMutator_4</mutator>
    <description>RemoveSwitch 4 mutation</description>
    <lineContent>switch (ast.getType()) {</lineContent>
  </mutation>
```

Comment (nick-mancuso) [reply .ci/pitest-suppressions/pitest-blocks-suppressions.xml:144]: Very interesting:

```

➜  checkstyle git:(master) ✗ git diff 
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java b/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java
index db864b415..c16a02b2f 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java
@@ -551,18 +551,18 @@ public class RightCurlyCheck extends AbstractCheck {
             switch (ast.getType()) {
                 case TokenTypes.LITERAL_TRY:
                 case TokenTypes.LITERAL_CATCH:
-                case TokenTypes.LITERAL_FINALLY:
                     details = getDetailsForTryCatchFinally(ast);
                     break;
                 case TokenTypes.LITERAL_IF:
-                case TokenTypes.LITERAL_ELSE:
                     details = getDetailsForIfElse(ast);
                     break;
                 case TokenTypes.LITERAL_DO:
-                case TokenTypes.LITERAL_WHILE:
-                case TokenTypes.LITERAL_FOR:
                     details = getDetailsForLoops(ast);
                     break;
+                case TokenTypes.LITERAL_FOR:
+                case TokenTypes.LITERAL_WHILE:
+                case TokenTypes.LITERAL_ELSE:
+                case TokenTypes.LITERAL_FINALLY:
                 default:
                     details = getDetailsForOthers(ast);
                     break;

➜  checkstyle git:(master) ✗ mvn test -Dtest=RightCurlyCheckTest test |& grep "Tests run"
[INFO] Tests run: 38, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.273 s - in com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheckTest
[INFO] Tests run: 38, Failures: 0, Errors: 0, Skipped: 0

```

**Review by nick-mancuso — APPROVED** [review]

**Comment by romani** [comment]
Single review is fine, this is very co technical update, no changes in main logic

**Pull request merged**